typesetters/base.lua

```diff
@@ -1,3 +1,81 @@
+--- SILE typesetter (default/base) class.
```

Comment:
How about putting these back in the `class:declareSettings()` but redefining the `SILE.settings:declare()` function to check for existing settings of the same name and warning on collisions and *NOT* doing anything, just skipping the duplicate declaration?

(Note they were in `class:_init()` which is wrong, they should be in `class:declareSettings()` as a future-proofing against scope refactors...)

Reply:
I'll take another look, but my gut reaction is to say that *correctness* is far more important than *efficiency* for this issue. We're about to try to untangle a giant web of cross wired setting scopes and I'm much more concerned about getting them mapped logically than how many typesetters are instanciated. If there is a speed problem (say with `ptable`) then you could instantiate one typesetter, then clone it for each new cell with similar properties.

Reply:
On second look I'm still of the same mind. I am deeply concerned about the speed of SILE (it should be 100× faster, multi threaded, and support resuming from the last change, not just little optimizations here and there) but the real way to fix that is to rewrite some bits of it in Rust. More than 95% of our time is spent iterating over node structures over and over in a way that Rust could do several orders of magnitude faster.

But I can't even think about making that happen until we have same separation of concerns. Until the typesetter/frame/class/package/pagebuilder/shaper/outputter interfaces have consistent inputs and outputs and don't rely on side effects (like global variables being changed when loading modules rather than on initializing them) I can't start swapping out pieces for higher performance.

Reply:
Ya not *executing* code in any modules when they load is going to be super important, everything needs to be return as a table or function, no side effects of `require()` itself...

I should probably setup a CI check for that.